lhotse/dataset/webdataset.py

```diff
@@ -126,6 +117,129 @@ def export_to_webdataset(
     return num_shards_written
 
 
+class WebdatasetWriter:
+    """
+    Saves the CutSet metadata along with audio/features data into a WebDataset archive.
+    The audio and feature data is read, decoded, and encoded into ``audio_format`` for audio,
+    lilcom for features and arrays with floating point type, and pickle for all other dtypes.
+    The intended use of this function is to speed up the I/O in training data pipelines by
+    converting random access reads to sequential access reads.
+
+    Supported values for ``audio_format`` are the same as for the ``format`` argument in
+    ``torchaudio.save`` function with ``sox_io`` backend.
+
+    If ``shard_size`` is specified, we will leverage WebDataset's ``ShardWriter`` to
+    create multiple tarballs with ``shard_size`` items per shard. In that mode, we expect
+    that ``output_path`` contains a pattern like "/path/to/shard-%06d.tar", which will
+    be internally expanded with the shard index.
+
+    Returns number of written shards if sharding is enabled, otherwise 0.
+
+    .. note: By default, we'll skip cuts which failed to load for any reason and proceed
+        with exporting. To raise an exception and stop, set ``fault_tolerant=False``.
+
+    **Example**
+
+    Export cuts with audio, features, and all custom data to a tarball shards with 500
+    cuts each::
+
+        >>> cuts = CutSet.from_jsonl_lazy("data/cuts-train.jsonl")
+        >>> with WebdatasetWriter("data/tars/shard-%06d.tar", shard_size=500) as writer:
+        ...     for cut in cuts:
+        ...         writer.write(cut)
+        >>> output_paths = writer.output_manifest_paths()
+
+    See also: :func`.export_to_webdataset`
```

Comment:
For this and the other web-dataset writing functions, it would be nice to have an indication to how one is supposed to read this data.  Since only the tar archive is output, it's not clear to me how one would grab its metadata.